cli/runtime/config/cli_discovery_sources_it_test.go

```diff
@@ -115,43 +122,75 @@ contexts:
         required: true
         annotationStruct:
             one: one
-        endpoint: updated-test-endpoint
-        path: updated-test-path
-        context: updated-test-context
+        endpoint: test-endpoint
+        path: test-path
+        context: test-context
       discoverySources:
         - gcp:
             name: test
-            bucket: updated-test-bucket
-            manifestPath: updated-test-manifest-path
+            bucket: test-bucket
+            manifestPath: test-manifest-path
             annotation: one
             required: true
           contextType: tmc
         - gcp:
             name: test-two
-            bucket: updated-test-bucket
-            manifestPath: updated-test-manifest-path
+            bucket: test-bucket
+            manifestPath: test-manifest-path
             annotation: two
             required: true
           contextType: tmc
 currentContext:
     k8s: test-mc
 `
-	return tanzuConfigBytes, expectedConfig
+
+	return CFG, expectedCFG, CFG2, expectedCFG2
 }
 
 func TestCLIDiscoverySourceIntegration(t *testing.T) {
-	// Setup Data and Test config file
-	tanzuConfigBytes, expectedConfig := setupData()
-	f, err := os.CreateTemp("", "tanzu_config")
+	// Setup config data
+	cfg, expectedCfg, cfg2, expectedCfg2 := setupData()
+
+	f1, err := os.CreateTemp("", "tanzu_config")
 	assert.Nil(t, err)
-	err = os.WriteFile(f.Name(), []byte(tanzuConfigBytes), 0644)
+	err = os.WriteFile(f1.Name(), []byte(cfg), 0644)
 	assert.Nil(t, err)
+
```

Comment:
nit: Can we refactor this config setup/cleanup code that seems to be identical or very similar across more than a dozen places? It would reduce the amount of test changes considerably. You can have the setup return a cleanup function that you can use to control when to call as appropriate in your test.

Reply:
refactored the test data helper functions